helper/Data.php

```diff
@@ -128,9 +135,16 @@ public function fetchPaymentMethods(Cart $cart, $language)
         $amount = $cart->getOrderTotal();
         $currencyData = Currency::getCurrency($cart->id_currency);
         $currency = $currencyData['iso_code'];
-        $address = new Address($cart->id_address_invoice);
-        $countryCode = Country::getIsoById($address->id_country);
         $shopperLocale = $this->languageAdapter->getLocaleCode($language);
+        $address = new Address($cart->id_address_invoice);
+
+        // If address is not created correctly and guest and one-page checkout are enabled, use context country
+        if (is_null($address->id_country) && $this->areGuestAndOnePageCheckout16()) {
```

Comment:
does it depend on the `$this->areGuestAndOnePageCheckout16()` result or anly on the `$address` object if it does exist or not?